contracts/AssetPoolFactory/AssetPoolFactoryFacet.sol

```diff
@@ -31,13 +31,17 @@ contract AssetPoolFactoryFacet is IAssetPoolFactory {
      * @notice Deploys and stores the reference to an asset pool based on the current defaultCut.
      * @dev Transfers ownership to the controller and initializes access control.
      * @param _facets Asset Pool facets for the factory diamond to deploy.
+     * @param _registry Registry address to point the pool to.
      */
-    function deployAssetPool(IDiamondCut.FacetCut[] memory _facets) external override {
+    function deployAssetPool(IDiamondCut.FacetCut[] memory _facets, address _registry) external override {
         LibDiamond.enforceIsContractOwner();
         LibFactoryStorage.Data storage s = LibFactoryStorage.s();
         //direct is required for the initialize functions below
         RelayDiamond d = new RelayDiamond(_facets, address(this));
         IDefaultDiamond assetPool = IDefaultDiamond(address(d));
+        if (_registry != address(0)){
```

Comment:
@bram-rongen What was the reason again to not just do this for every deployed pool? Modularity?

Reply:
Moving forward with forced registry configuration on deploy. We should probably solve the current modularity issues in this method in a different manner.